BLE Indoor Positioning/src/main/java/com/nexenio/bleindoorpositioning/ble/beacon/BeaconManager.java

```diff
@@ -59,11 +63,13 @@ public static void processAdvertisingPacket(String macAddress, AdvertisingPacket
 
     public static void processClosestBeacon(Beacon beacon) {
         if (instance.closestBeacon == null) {
-            instance.setClosestBeacon(beacon);
+            instance.closestBeacon = beacon;
         } else {
-            if (instance.getClosestBeacon().getMacAddress() != beacon.getMacAddress()){
-                if (beacon.getDistance() < instance.closestBeacon.getDistance()) {
-                    instance.setClosestBeacon(beacon);
+            if (instance.closestBeacon != beacon){
+                if(beacon.getLatestAdvertisingPacket().getTimestamp() > System.currentTimeMillis() - TimeUnit.SECONDS.toMillis(10)){
```

Comment:
This will always be true, as you call `processClosestBeacon` from `processAdvertisingPacket`.
Also, use the formatter before committing.